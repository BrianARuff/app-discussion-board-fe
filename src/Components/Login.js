import React from "react";
import { Link } from "react-router-dom";
import { InputLabel, FormHelperText, Input, FormControl, Button } from "@material-ui/core";
import axios from "axios";
import Cookie from "js-cookie";
import "../styles/loggin.css";

export default class Login extends React.Component {
  state = {
    name: "",
    email: "",
    password: "",
    status: "Register"
  }

  handleInputChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  loginUser = () => {
    const { name, email, password } = this.state;
    this.setState({status: "Validating Credentials"});
    axios.post("https://suicide-watch-backend.herokuapp.com/auth/login", {name, email, password})
      .then(res => {
        Cookie.set("token", res.data.token, {
          expires: 7,
          path: "/"
        });
        this.setState({status: "Authentication successful"});
        this.props.history.push('/');
        window.location.reload();
      })
      .catch(err => {
        console.error(this.status, err);
        this.setState({status: "Failed to authenticate your account please try again..."});
      });
  }

  render() {
    return (
      <div className="login-form-container">
        <h1 className="hide">Login Page</h1>
        <h2 className="fw-bold">Login Below</h2>
        <h4 className="fw-light">Fill out either your email <b>or</b> username, not both.</h4>
        <div className="login-form-flex">
          <FormControl fullWidth required>
            <InputLabel>
              Username
            </InputLabel>
            <Input name="name" type="text" onChange={this.handleInputChange} />
            <FormHelperText>Fill out either your email or password.</FormHelperText>
          </FormControl>

          <FormControl fullWidth required>
            <InputLabel>
              Email
            </InputLabel>
            <Input name="email" type="email" onChange={this.handleInputChange} />
            <FormHelperText>Fill out either your email or password</FormHelperText>
          </FormControl>

          <FormControl fullWidth required>
            <InputLabel>
              Password
            </InputLabel>
            <Input name="password" type="password" onChange={this.handleInputChange} />
            <FormHelperText>Please enter a strong password that you can remember</FormHelperText>
          </FormControl>

          <br />
          <FormControl fullWidth>
            <Button
              onClick={this.loginUser}
              fullWidth
              color="secondary"
              variant="contained"
            >
              {this.state.status}
            </Button>
            <FormHelperText>Not yet a member? That is okay! Please head over to the <Link to="/register">Register</Link> page to register your new account!!</FormHelperText>
          </FormControl>
        </div>
      </div>
    )
  }
}