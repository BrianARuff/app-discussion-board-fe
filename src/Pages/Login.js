import React, { Fragment } from "react";
import {
  InputLabel,
  FormHelperText,
  Input,
  FormControl,
  Button
} from "@material-ui/core";
import axios from "axios";
import Cookie from "js-cookie";
import ClipLoader from "react-spinners/ClipLoader";
import "../styles/loggin.css";

export default class Login extends React.Component {
  state = {
    name: "",
    email: "",
    password: "",
    status: "Login"
  };

  handleInputChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  loginUser = () => {
    const { name, email, password } = this.state;
    this.setState({ status: "Validating Credentials" });
    axios
      .post("https://suicide-watch-backend.herokuapp.com/auth/login", {
        name,
        email,
        password
      })
      .then(res => {
        Cookie.set("token", res.data.token, {
          expires: 7,
          path: "/"
        });
        this.setState({ status: "Authentication successful" });
        this.props.history.push("/");
        window.location.reload();
        window.scrollTo({
          top: 0,
          behavior: "auto"
        });
      })
      .catch(err => {
        console.error(this.status, err);
        this.setState({
          status: "Failed to authenticate your account please try again..."
        });
      });
  };

  loginUserEnter = e => (e.key === "Enter" ? this.loginUser() : null);

  render() {
    return (
      <div className="login-form-container">
        <h1 className="hide">Login Page</h1>
        <h2 className="fw-bold">Login Below</h2>
        <h4 className="fw-light">
          Login with either your email or username in the appropriate field
        </h4>
        <form
          onKeyDown={e => this.loginUserEnter(e)}
          className="login-form-flex"
        >
          <FormControl fullWidth required>
            <InputLabel>Username</InputLabel>
            <Input
              autoComplete="true"
              name="name"
              type="text"
              onChange={this.handleInputChange}
            />
            <FormHelperText>Fill out either your username</FormHelperText>
          </FormControl>

          <FormControl fullWidth required>
            <InputLabel>Email</InputLabel>
            <Input
              autoComplete="true"
              name="email"
              type="email"
              onChange={this.handleInputChange}
            />
            <FormHelperText>Fill out either your email</FormHelperText>
          </FormControl>

          <FormControl fullWidth required>
            <InputLabel>Password</InputLabel>
            <Input
              autoComplete="true"
              name="password"
              type="password"
              onChange={this.handleInputChange}
            />
            <FormHelperText>Please enter your password</FormHelperText>
          </FormControl>

          <br />
          <FormControl fullWidth>
            <Button
              onClick={this.loginUser}
              fullWidth
              color="secondary"
              variant="contained"
            >
              {this.state.status === "Validating Credentials" ? (
                <Fragment>
                  <span style={{ position: "relative", top: 0, right: 10 }}>
                    <ClipLoader />
                  </span>
                  {this.state.status}
                </Fragment>
              ) : (
                this.state.status
              )}
            </Button>
            <FormHelperText>
              Don't have an account? Head over to the{" "}
              <a href="/register">Register</a> page to register your new
              account!!
            </FormHelperText>
          </FormControl>
        </form>
      </div>
    );
  }
}
