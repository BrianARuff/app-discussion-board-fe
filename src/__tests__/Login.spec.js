import React from "react";
import { create } from "react-test-renderer";
import Login from "../Pages/Login";

describe("test Login component", () => {
  test("Login snapshot from json", () => {
    const login = create(<Login />);
    expect(login.toJSON()).toMatchSnapshot();
  });
});
