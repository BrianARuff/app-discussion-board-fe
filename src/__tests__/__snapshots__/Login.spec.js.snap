// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Login component Login snapshot from json 1`] = `
<div
  className="login-form-container"
>
  <h1
    className="hide"
  >
    Login Page
  </h1>
  <h2
    className="fw-bold"
  >
    Login Below
  </h2>
  <h4
    className="fw-light"
  >
    Login with either your email or username in the appropriate field
  </h4>
  <form
    className="login-form-flex"
    onKeyDown={[Function]}
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
      >
        Username
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-required"
      >
        Fill out either your username
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
      >
        Email
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-required"
      >
        Fill out either your email
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
      >
        Password
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="true"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
        />
      </div>
      <p
        className="MuiFormHelperText-root Mui-required"
      >
        Please enter your password
      </p>
    </div>
    <br />
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
      <p
        className="MuiFormHelperText-root"
      >
        Don't have an account? Head over to the
         
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
         
        page to register your new account!!
      </p>
    </div>
  </form>
</div>
`;
